pkg/operator/controller/controller_dns_daemonset_test.go

```diff
@@ -469,14 +481,23 @@ func TestDaemonsetConfigChanged(t *testing.T) {
 			expect: true,
 		},
 		{
-			description: "if the update strategy changes",
+			description: "if the update strategy's max unavailable parameter changes",
 			mutate: func(daemonset *appsv1.DaemonSet) {
-				daemonset.Spec.UpdateStrategy = appsv1.DaemonSetUpdateStrategy{
-					Type: appsv1.RollingUpdateDaemonSetStrategyType,
-					RollingUpdate: &appsv1.RollingUpdateDaemonSet{
-						MaxUnavailable: pointerTo(intstr.FromString("10%")),
-					},
-				}
+				daemonset.Spec.UpdateStrategy.RollingUpdate.MaxUnavailable = pointerTo(intstr.FromString("10%"))
+			},
+			expect: true,
+		},
+		{
+			description: "if the update strategy's max surge parameter changes",
+			mutate: func(daemonset *appsv1.DaemonSet) {
+				daemonset.Spec.UpdateStrategy.RollingUpdate.MaxSurge = pointerTo(intstr.FromString("10%"))
```

Comment:
I'm confused, this doesn't change the default value?

Reply:
L495 `expect: true` means it expects a change

Reply:
The old value doesn't specify `maxSurge`: https://github.com/openshift/cluster-dns-operator/blob/b0ca398b84de1287fe02d22bf9311f0053a02d1c/pkg/operator/controller/controller_dns_daemonset_test.go#L623-L628

Reply:
So, the assets/dns/daemonset.yaml specifies `maxSurge`, but that isn't in a new `DaemonSetUpdateStrategy`?

Reply:
`maxSurge` isn't in the daemonset that is used in the test.  The daemonset that `TestDaemonsetConfigChanged` uses differs from the daemonset that `desiredDNSDaemonSet` returns.